Split crontab.yml in to base and busy files

michaeljcollinsuk opened the pull request:
- Base file should always be included
- Busy file contains jobs for busy times around planned elections in
May etc

symroe commented:
This looks like the right approach!

I think the following should be imported all the time:

* Recent people
* Leaflets
* Slack

But ideally not as often?

I see a couple of ways of doing this:

1. only include a single file ever, where we duplicate the commands between each, and tweak the times
2. Implement something like the ["rest mode" server in WDIV](https://github.com/DemocracyClub/polling_deploy/commit/f9848c0a1389394aded7b85fc4b0f567faaf763c) to pass a var in to Ansible that [we can use later](https://github.com/DemocracyClub/polling_deploy/commit/56e8a2a701ceddd26492a7242b2a495b46691296#diff-da14ad81019768f718a0940f8c992f5a8d28f0d7336928c24796d0a18e12c9fdR22)


MVP is just to add them back to the main crontab file and make a note to change the times later!

michaeljcollinsuk commented:
> This looks like the right approach!
> 
> I think the following should be imported all the time:
> 
> * Recent people
> * Leaflets
> * Slack
> 
> But ideally not as often?
> 
> I see a couple of ways of doing this:
> 
> 1. only include a single file ever, where we duplicate the commands between each, and tweak the times
> 2. Implement something like the ["rest mode" server in WDIV](https://github.com/DemocracyClub/polling_deploy/commit/f9848c0a1389394aded7b85fc4b0f567faaf763c) to pass a var in to Ansible that [we can use later](https://github.com/DemocracyClub/polling_deploy/commit/56e8a2a701ceddd26492a7242b2a495b46691296#diff-da14ad81019768f718a0940f8c992f5a8d28f0d7336928c24796d0a18e12c9fdR22)
> 
> MVP is just to add them back to the main crontab file and make a note to change the times later!

I've gone with MVP and added those three back in to the `base` file (and also moved the referendums job) so that they will still run if you are still planning a deploy later. And can work on better solutions later, as I know we already had some discussion about changing how crons are run more generally (we were thinking about defining them within the project repo I think?).

Also have tweaked the times of leaflets and recent people jobs to:
- leaflets runs once daily at 00:35
- recent people every 4 hours, at 45 mins past the hour (6 runs daily in total). 

Although just to double check as I may have misunderstood something. The reason I had moved the recent people cron to the 'busy' file was I thought you had mentioned that quieter times, the `import_ballots --current` job would be sufficient (as part of this is it updates candidacies)?

michaeljcollinsuk commented:
Also just noticed that `import_ballots --current` is [set to 'disabled'](https://github.com/DemocracyClub/who_deploy/blob/master/crontab.yml#L43) - should it be?

symroe commented:
> Also just noticed that `import_ballots --current` is [set to 'disabled'](https://github.com/DemocracyClub/who_deploy/blob/master/crontab.yml#L43) - should it be?

I think I disabled this over 2020 - the idea is to import ballots on the same day as making them in EE->YNR, especially for by-elections.

Problem is: when we have loads of current elections it takes ages.

Maybe we need to move (in all data we import!) to a model where we save the last updated timestamp of each election|person|etc and then just and YNR for any data updated since the last thing we saw? This is [how the `--recent` flag works on `import_people`](https://github.com/DemocracyClub/WhoCanIVoteFor/blob/master/wcivf/apps/people/management/commands/import_people.py#L54).

symroe commented:
(All these commands could run every 5 minutes or however frequently and it wouldn't matter too much)

symroe reviewed (approved):
I think in board terms this is good to go. We can tweak the exact times of each command, but I'm happy to merge and deploy this before getting into that if you like. 

Up to you as to how you want to go order the work though - feel free to use this PR to change the times / update the commands

michaeljcollinsuk commented:
@symroe I really like the idea of storing `last_updated` timestamps for everything and then moving to that model of importing `recent` changes. Then in theory the jobs should always be quite small, which would mean we need to worry less about the timings and spacing jobs out.

As for right now, I'm happy to merge so that it can be deployed to see if [helps resolve those other errors](https://github.com/DemocracyClub/EveryElection/issues/1119).
